### ISSUE TITLE
Blog: Techniques for training large neural networks

### ISSUE THREAD

**Issue by konabuta** [konabuta/paper-survey#14]
## Blog: Techniques for training large neural networks
Link: https://openai.com/research/techniques-for-training-large-neural-networks

This blog explains the parallism technologies for training large neural networks.
![image](https://github.com/konabuta/my-survey/assets/3815677/44980868-562d-4c16-8df9-5ad586561921)

**Comment by konabuta** [comment]
## Data parallelistm

1. indendently compte the gradient on each worker.
2. average the gradents across workers.
3. independently compute the same new parameters on each worker.

Step can be bottleneck becuase it requires transferring data.

**Comment by konabuta** [comment]
## Pipeline parallelism
- bubbls: a worker waits for outputs from the previous machine to be used as its inputs.
![image](https://github.com/konabuta/my-survey/assets/3815677/698da8dd-3b64-4f5e-ac43-184ec6165fee)

**Comment by konabuta** [comment]
## Tensor parallelism

**Comment by konabuta** [comment]
## Mixture-of-Experts (MoE)

**Comment by konabuta** [comment]
## Others
- Checkpointing
- Mixed Precision Training
- Offloading
- Memory Efficient Optimizer
- Compression

**Comment by konabuta** [comment]
## Reference
- https://developer.nvidia.com/blog/mastering-llm-techniques-inference-optimization/
